### PR TITLE
CI: xfail geopandas downstream test on MacOS due to fiona install

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Not really familiar with geopandas to know why this install fails on MacOS, so xfailing for now (maybe there's a more lightweight test that doesn't read a file?)

cc @jorisvandenbossche  

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=75056&view=logs&jobId=b1891a80-bdb0-5193-5f59-ce68a0874df0&j=b1891a80-bdb0-5193-5f59-ce68a0874df0&t=0a58a9e9-0673-539c-5bde-d78a5a3b655e

```
=================================== FAILURES ===================================
________________________________ test_geopandas ________________________________
[gw0] darwin -- Python 3.9.10 /usr/local/miniconda/envs/pandas-dev/bin/python

    @pytest.mark.filterwarnings("ignore:can't resolve:ImportWarning")
    def test_geopandas():
    
        geopandas = import_module("geopandas")
        fp = geopandas.datasets.get_path("naturalearth_lowres")
>       assert geopandas.read_file(fp) is not None

pandas/tests/test_downstream.py:203: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/miniconda/envs/pandas-dev/lib/python3.9/site-packages/geopandas/io/file.py:166: in _read_file
    _check_fiona("'read_file' function")
E           ImportError: the 'read_file' function requires the 'fiona' package, but it is not installed or does not import correctly.
E           Importing fiona resulted in: dlopen(/usr/local/miniconda/envs/pandas-dev/lib/python3.9/site-packages/fiona/ogrext.cpython-39-darwin.so, 2): Library not loaded: @rpath/libfontconfig.1.dylib
E             Referenced from: /usr/local/miniconda/envs/pandas-dev/lib/libpoppler.117.0.0.dylib
E             Reason: Incompatible library version: libpoppler.117.dylib requires version 14.0.0 or later, but libfontconfig.1.dylib provides version 13.0.0

```

### CODE REVIEW

**Comment by jbrockmendel** [comment]
This test passes for me locally on MacOS with py3.9.10.  Maybe it's just a py310 thing?

**Comment by jorisvandenbossche** [comment]
This is a conda-forge packaging hickup, was going to open an issue but they are already aware / trying to fix it: https://github.com/conda-forge/fontconfig-feedstock/issues/54 / https://github.com/conda-forge/fontconfig-feedstock/pull/53#issuecomment-1062456875

> This test passes for me locally on MacOS with py3.9.10. Maybe it's just a py310 thing?

It shouldn't be related to the python version, I would think (since it's a C dependency), but note that it's only an issue with the latest version of the package from conda-forge on Mac (so would need to have created such an environment the last day to see the issue locally-

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

I don't think we need to put too much effort in fine tuning the exact skip, since we will be able to remove it again in a few days if all the packages have been rebuilt.

File: pandas/tests/test_downstream.py [hunk 1]
```diff
@@ -196,6 +201,13 @@ def test_pandas_datareader():
 
 # importing from pandas, Cython import warning
 @pytest.mark.filterwarnings("ignore:can't resolve:ImportWarning")
+@pytest.mark.xfail(
+    is_platform_mac() and (PY39 or PY310),
```
Comment (Dr-Irv) [inline pandas/tests/test_downstream.py:205]: I saw a failure in python 3.8 test on MacOS as well.

**Comment by mroeschke** [comment]
Looks like on the latest main run this affects 3.9 and 3.10:

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=75056&view=logs&jobId=b1891a80-bdb0-5193-5f59-ce68a0874df0&j=b1891a80-bdb0-5193-5f59-ce68a0874df0

**Comment by mroeschke** [comment]
Do you recommend a backport for this @jorisvandenbossche or should we not bother since it should be fixed in the next couple of days?

**Comment by jorisvandenbossche** [comment]
That probably depends on the timeline of 1.4.2. Although I would assume this should be fixed by now, we can check by removing the skip again.

**Comment by jorisvandenbossche** [comment]
-> https://github.com/pandas-dev/pandas/pull/46320

**Pull request merged**